crates/storage/codecs/src/alloy/eip4844.rs

```diff
@@ -0,0 +1,74 @@
+use crate::Compact;
+use alloy_consensus::transaction::TxEip4844 as AlloyTxEip4844;
+use alloy_eips::eip2930::AccessList;
+use alloy_primitives::{Address, Bytes, ChainId, B256, U256};
+use reth_codecs_derive::add_arbitrary_tests;
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+
+/// [EIP-4844 Blob Transaction](https://eips.ethereum.org/EIPS/eip-4844#blob-transaction)
+///
+/// This is a helper type to use derive on it instead of manually managing `bitfield`.
+///
+/// By deriving `Compact` here, any future changes or enhancements to the `Compact` derive
+/// will automatically apply to this type.
+///
+/// Notice: Make sure this struct is 1:1 with [`alloy_consensus::transaction::TxEip4844`]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default, Serialize, Deserialize, Compact)]
+#[cfg_attr(test, derive(arbitrary::Arbitrary))]
+#[add_arbitrary_tests(compact)]
+struct TxEip4844 {
```

Comment:
we had the following field here before: https://github.com/paradigmxyz/reth/blob/24f04710c0d94bb61db65f90d166ef1c8d39a5d6/crates/primitives/src/transaction/eip4844.rs#L65-L68

And this PR removes it, I think it will be a breaking change for DB and we should add it back?

Reply:
I just wanted to get some feedback on this. I don't have enough perspective to say how to deal with it:
- I didn't think it was useful (but I'm probably wrong).
- I have the impression that all the tests pass but maybe it's a false indicator for that.
- If we have to add it, we should also put it in alloy and that calls on `CompactPlaceholder` which implements `Compact` so we can't really add it to alloy right?

Reply:
My understanding here is that we don't really care about value of that field and only need it to exist on `Compact` helper. It is probably safe and correct to always set it to `Some(())` to preserve backwards compatibility with older reth versions which will attempt to read it into `TxKind`. Thus, there's no need in adding it to alloy

however would like to get confirmation from @joshieDo on this

Reply:
I just put the field back in our private `TxEip4844` structure to use it only in the `Compact` implementation as `Some(())` as a default

Reply:
if it can be removed, we should open a separate issue for this

Reply:
Tests should be put back